xla/service/gpu/nccl_api.h

```diff
@@ -191,6 +196,7 @@ class NcclApi {
                             PrimitiveType dtype, size_t count, int32_t peer,
                             NcclCommHandle comm, se::Stream* stream) = 0;
 
+  #ifdef XCCL_HAS_COMM_REGISTER
```

Comment:
I’d love to avoid adding new ifdefs to headers, can this be just empty function in cc, or return an error? These iddefs tend to spread and make life complicated. The plan is to provide single header that can be included and used anywhere in XLA